### PR TITLE
Use `<kbd>` in documentation

### PR DESCRIPTION
**Pull request by Meakk**
Preview can be found here: https://github.com/Meakk/f3d/blob/kbd-html/doc/user/INTERACTIONS.md

### CODE REVIEW

**Review by mwestphal — APPROVED** [review]

**Comment by Meakk** [comment]
I'll merge it but I'm not sure "just the docs" have the proper css styling for it.

**Comment by mwestphal** [comment]
>  "just the docs" have the proper css styling for it.

You can merge and fix it after if needed, or you can deploy the website on your fork if you want.

**Comment by Meakk** [comment]
> or you can deploy the website on your fork if you want.

How can I do that?

**Comment by Meakk** [comment]
That looks bad on the live website
image

I'll see if I can change the CSS later. I'm putting this PR on hold for now.

**Comment by Meakk** [comment]
2024-01-09-225056_1286x1090

I have added a custom css file picked up automatically by just-the-docs.

File: _sass/custom/custom.scss [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    margin-right: 3px;
+    position: relative;
+    top: -1px;
+
+    // Push effect on hover
+    &:hover {
+        box-shadow: 0 1px 0 0.5px var(--kbd-color-border);
+        top: 1px;
+    }
+}
```
Comment (mwestphal) [inline _sass/custom/custom.scss:35]: I had no idea you were a magician @Meakk

**Review by mwestphal — APPROVED** [review]

File: _sass/custom/custom.scss (hunk 1 above)
Comment (Meakk) [reply _sass/custom/custom.scss:35]: 🧙‍♂️

File: _sass/custom/custom.scss [hunk 2]
```diff
@@ -0,0 +1,35 @@
+kbd {
+    // TODO: use just the docs variables?
+    --kbd-color-background: #e6e1e8;
+    --kbd-color-border: #959396;
+    --kbd-color-text: #27262b;
+
+    // Match the page defaults in light mode but
+    // switch to a light-ish grey in dark mode
```
Comment (snoyer) [inline _sass/custom/custom.scss:8]: Not sure what this comment is referring to as there's no `prefers-color-scheme` stuff around.
Does the F3D website have a light mode or is it dark only?

File: _sass/custom/custom.scss [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    line-height: 1;
+    min-width: 0.75rem;
+    text-align: center;
+    padding: 2px 5px;
+    margin-right: 3px;
+    position: relative;
+    top: -1px;
+
+    // Push effect on hover
+    &:hover {
```
Comment (snoyer) [inline _sass/custom/custom.scss:31]: If you want to add to the effect:
```suggestion
    &:hover, li:hover & {
```
`li:hover kbd` matches `<kbd>` elements inside hovered list items, so the keys would also get pressed when the cursor is over the corresponding description text

File: doc/user/INTERACTIONS.md [hunk 4]
```diff
@@ -18,55 +18,55 @@ Simple interaction with the displayed data is possible directly within the windo
...
+* <kbd>N</kbd>: the display of the file name.
+* <kbd>M</kbd>: the display of the metadata if exists.
+* <kbd>Z</kbd>: the display of the FPS counter.
+* <kbd>R</kbd>: raytracing.
+* <kbd>D</kbd>: the denoiser when raytracing.
+* <kbd>U</kbd>: background blur, useful with an HDRI skybox.
+* <kbd>K</kbd>: trackball interaction mode.
+* <kbd>F</kbd>: HDRI ambient lighting.
+* <kbd>J</kbd>: the display of the HDRI skybox.
+* <kbd>L</kbd>: increase (+Shift: decrease) lights intensity.
```
Comment (snoyer) [inline doc/user/INTERACTIONS.md:49]:
```suggestion
* <kbd>L</kbd>: increase (<kbd>Shift</kbd>+<kbd>L</kbd>: decrease) lights intensity.
```
or
```suggestion
* <kbd>L</kbd>: increase lights intensity.
* <kbd>Shift</kbd>+<kbd>L</kbd>: decrease lights intensity.
```
maybe? Not sure what's best here

File: _sass/custom/custom.scss (hunk 2 above)
Comment (Meakk) [reply _sass/custom/custom.scss:8]: I copy-pasted a random code from internet and realized it was not needed but forgot to remove the comment ^^

File: _sass/custom/custom.scss (hunk 3 above)
Comment (Meakk) [reply _sass/custom/custom.scss:31]: interesting! Thanks

**Comment by Meakk** [comment]
@snoyer please take another look.  
Deployed page is here: https://meakk.github.io/f3d/doc/user/INTERACTIONS.html

**Comment by snoyer** [comment]
Looks good! not sure the `li:hover` is useful but it's fun 

The only thing that I find not amazing now is that the arrow keys looks tiny compared to the others :( is it just me?
Unicode has a few bigger ones but I don't know if that's asking for compatibility issues: `🡨 🡪 🡩 🡫` (comparison: <kbd>🡨</kbd><kbd>&larr;</kbd>)

**Comment by mwestphal** [comment]
a

🤣

**Comment by snoyer** [comment]
@mwestphal I'm sure glad I qualified my statement twice then!
So... my suggestion (arrows) _does_ fail bad at compatibility! but it is not just me, that arrow looks tiny on your setup too

**Comment by Meakk** [comment]
Let's stick for regular HTML ampersand aliases, using HTML is asking for troubles.

**Pull request merged**